operator/iamkeycontroller/webhook_test.go

```diff
@@ -11,165 +11,150 @@ import (
-					ResourceSpec: xpv1.ResourceSpec{
-						ProviderConfigReference: &xpv1.Reference{
-							Name: "provider-config",
-						},
-						WriteConnectionSecretToReference: &xpv1.SecretReference{
-							Name:      "secret-name",
-							Namespace: "secret-namespace",
-						},
-					},
-					ForProvider: exoscalev1.IAMKeyParameters{
-						Services: exoscalev1.ServicesSpec{},
-					},
-				},
-			},
+			bucketNames:   []string{},
 			expectedError: "an IAMKey named \"iamkey-with-no-buckets\" should have at least 1 allowed bucket",
 		},
 		"GivenIAMKey_When2Buckets_ThenExpectNoError": {
-			iamKey: exoscalev1.IAMKey{
-				ObjectMeta: metav1.ObjectMeta{
-					Name: "iamkey-with-buckets",
-				},
-				Spec: exoscalev1.IAMKeySpec{
-					ResourceSpec: xpv1.ResourceSpec{
-						ProviderConfigReference: &xpv1.Reference{
-							Name: "provider-config",
-						},
-						WriteConnectionSecretToReference: &xpv1.SecretReference{
-							Name:      "secret-name",
-							Namespace: "secret-namespace",
-						},
-					},
-					ForProvider: exoscalev1.IAMKeyParameters{
-						Services: exoscalev1.ServicesSpec{
-							SOS: exoscalev1.SOSSpec{
-								Buckets: []string{"bucket.1", "bucket.2"},
-							},
-						},
-					},
-				},
-			},
+			bucketNames:   []string{"bucket.1", "bucket.2"},
 			expectedError: "",
 		},
-		"GivenIAMKey_WhenWriteConnectionSecretToRefNameAndNamespaceExists_ThenExpectNoError": {
-			iamKey: exoscalev1.IAMKey{
-				ObjectMeta: metav1.ObjectMeta{
-					Name: "iamkey-with-buckets",
-				},
+	}
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			iamKey := exoscalev1.IAMKey{
+				ObjectMeta: metav1.ObjectMeta{Name: "iamkey-with-no-buckets"},
 				Spec: exoscalev1.IAMKeySpec{
 					ResourceSpec: xpv1.ResourceSpec{
-						ProviderConfigReference: &xpv1.Reference{
-							Name: "provider-config",
-						},
-						WriteConnectionSecretToReference: &xpv1.SecretReference{
-							Name:      "secret-name",
-							Namespace: "secret-namespace",
-						},
+						ProviderConfigReference:          &xpv1.Reference{Name: "provider-config"},
+						WriteConnectionSecretToReference: &xpv1.SecretReference{Name: "secret-name", Namespace: "secret-namespace"},
 					},
 					ForProvider: exoscalev1.IAMKeyParameters{
 						Services: exoscalev1.ServicesSpec{
 							SOS: exoscalev1.SOSSpec{
-								Buckets: []string{"bucket.1", "bucket.2"},
+								// buckets is being tested
+								Buckets: tc.bucketNames,
 							},
 						},
 					},
 				},
+			}
+			validator := &IAMKeyValidator{log: logr.Discard()}
+			err := validator.ValidateCreate(nil, &iamKey)
+			if tc.expectedError != "" {
+				assert.EqualError(t, err, tc.expectedError)
+			} else {
+				require.NoError(t, err)
+			}
+		})
+	}
+}
+
+func TestIAMKeyValidator_ValidateCreate_RequireWriteConnectionSecretToRef(t *testing.T) {
+	tests := map[string]struct {
+		iamKey                exoscalev1.IAMKey
+		connectionSecretToRef xpv1.SecretReference
```

Comment:
This should be a pointer, since the spec allows `nil`.

Also, currently there isn't a test case that tests that `WriteConnectionSecretToRef` is nil.
```suggestion
		connectionSecretToRef *xpv1.SecretReference
```